src/demo/OceanObject.js

```diff
@@ -0,0 +1,214 @@
+    const sideFinBack = this.sideFinsBack[finIdx];
+    const topFin = this.topFins[
+      Math.floor(Math.random() * this.topFins.length)
+    ];
+    const tail = this.tails[Math.floor(Math.random() * this.tails.length)];
+    this.colorPalette = this.colorPalettes[
+      Math.floor(Math.random() * this.colorPalettes.length)
+    ];
+    this.knnData = [
+      ...body.knnData,
+      ...eye.knnData,
+      ...mouth.knnData,
+      ...sideFinFront.knnData,
+      ...topFin.knnData,
+      ...tail.knnData,
+      ...this.colorPalette.knnData
+    ];
+    this.parts = [body, eye, mouth, sideFinFront, sideFinBack, topFin, tail];
+  }
+
+  getColorPalette() {
+    return this.colorPalette;
+  }
+
+  drawToCanvas(fishCanvas) {
+    const ctx = fishCanvas.getContext('2d');
+    ctx.translate(constants.fishCanvasWidth, 0);
+    ctx.scale(-1, 1);
+    const body = this.parts.find(part => part.type === FishBodyPart.BODY);
+    const bodyAnchor = bodyAnchorFromType(body, body.type);
+    const parts = _.orderBy(this.parts, ['type']);
+
+    parts.forEach((part, partIndex) => {
+      intermediateCtx.clearRect(
+        0,
+        0,
+        constants.fishCanvasWidth,
+        constants.fishCanvasHeight
+      );
+
+      let anchor = [0, 0];
+      if (part.type !== FishBodyPart.BODY) {
+        const bodyAnchor = bodyAnchorFromType(body, part.type);
+        anchor[0] = bodyAnchor[0];
+        anchor[1] = bodyAnchor[1];
+      }
+
+      const img = fishPartImages[part.type][part.index];
+
+      if (part.type === FishBodyPart.TAIL) {
+        anchor[1] -= img.height / 2;
+      }
+
+      const xPos = bodyAnchor[0] + anchor[0];
+      const yPos = bodyAnchor[1] + anchor[1];
+
+      intermediateCtx.drawImage(img, xPos, yPos);
+      const rgb = colorForFishPart(this.colorPalette, part);
+
+      if (rgb) {
+        // Add some random tint to the RGB value.
+        const tintAmount = 20;
+        let newRgb = [];
+        newRgb[0] = clamp(rgb[0] + randomInt(-tintAmount, tintAmount), 0, 255);
+        newRgb[1] = clamp(rgb[1] + randomInt(-tintAmount, tintAmount), 0, 255);
+        newRgb[2] = clamp(rgb[2] + randomInt(-tintAmount, tintAmount), 0, 255);
+
+        let imageData = intermediateCtx.getImageData(
+          xPos,
+          yPos,
+          img.width,
+          img.height
+        );
+        let data = imageData.data;
+
+        for (let i = 0; i < data.length; i += 4) {
+          // Tint any visible pixels
+          if (data[i + 3] > 0) {
+            data[i] = newRgb[0];
+            data[i + 1] = newRgb[1];
+            data[i + 2] = newRgb[2];
+          }
+        }
+
+        intermediateCtx.putImageData(imageData, xPos, yPos);
+      }
+      ctx.drawImage(
+        intermediateCanvas,
+        constants.fishCanvasWidth / 2 - intermediateCanvas.width / 2,
+        constants.fishCanvasHeight / 2 - intermediateCanvas.height / 2,
+        constants.fishCanvasWidth,
+        constants.fishCanvasHeight
```

Comment:
If we draw fish here, do we still need the `drawSingleFish` function in the renderer?

Reply:
oops, disregard